src/sdam/monitor.ts

```diff
@@ -279,6 +279,10 @@ function checkServer(monitor: Monitor, callback: Callback<Document | null>) {
       // if we are using the streaming protocol then we immediately issue another `started`
       // event, otherwise the "check" is complete and return to the main monitor loop
       if (isAwaitable && hello.topologyVersion) {
```

Comment:
I don't understand this conditional.  `isAwaitable` is initialized above with `monitor[kServer].description.topologyVersion != null`.  I think that is the description from the handshake.  But then we `and` that here with the new `hello.topologyVersion`.  For clarity I think that either `isAwaiting` should be removed from this conditional, or else we should just check `isAwaiting` and move the conditional up to line 235.

I'm not sure which is correct, or if both are correct, but there don't seem to be any tests that prove that  `connection.isStreamingProtocol` is ever set to `true`, so I have some cause for concern.

Perhaps it would be safer to just always set `connection.isStreamingProtocol` to `true` for all monitor connections.  It seems like that would be harmless (we could change the property is `isMonitorConnection`).

Reply:
I've updated per the suggestion to set the flag on all monitoring connections and added a unit test to the monitor to ensure its connection has the flag set. Property renamed to `isMonitoringConnection`.